modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java

```diff
@@ -1945,28 +1946,30 @@ private void validateIdentifier(String identifier, NodeLocation location) {
 
     /**
      * Validates the statements in the transform statement body as explained below :
-     *  - Left expression of Assignment Statement becomes output of transform statement
-     *  - Right expressions of Assignment Statement becomes input of transform statement
-     *  - Variables in each of left and right expressions of all statements are extracted as input and output
-     *  - A variable that is used as an input cannot be used as an output in another statement
-     *  - If inputs and outputs are used interchangeably, a semantic error is thrown
+     * - Left expression of Assignment Statement becomes output of transform statement
+     * - Right expressions of Assignment Statement becomes input of transform statement
+     * - Variables in each of left and right expressions of all statements are extracted as input and output
+     * - A variable that is used as an input cannot be used as an output in another statement
+     * - If inputs and outputs are used interchangeably, a semantic error is thrown
+     *
      * @param blockStmt transform statement block statement
-     * @param inputs input variable reference expressions map
-     * @param outputs output variable reference expressions map
+     * @param inputs    input variable reference expressions map
+     * @param outputs   output variable reference expressions map
      */
     private void validateTransformStatementBody(BlockStmt blockStmt, Map<String, Expression> inputs,
                                                 Map<String, Expression> outputs) {
         for (Statement statement : blockStmt.getStatements()) {
             if (statement instanceof AssignStmt) {
                 for (Expression lExpr : ((AssignStmt) statement).getLExprs()) {
-                    if (lExpr instanceof FieldAccessExpr) {
-                        String varName = ((FieldAccessExpr) lExpr).getVarName();
+                    Expression[] varRefExpressions = getVariableReferencesFromExpression(lExpr);
```

Comment:
what happens if I write a ballerina code like below?
transform {
func_A(name) = p.city;
}

Here the LHS is a function invocation statement. Isn't this an invalid statement?